### ISSUE TITLE
Cannot connect to SickRage

### ISSUE THREAD

**Issue by benwut** [MGaetan89/ShowsRage#17]
Got the message "Connection to http://192.168.1.100:8081/ failed". If I check the logs on SickRage I got a message:

2015-09-27 18:02:25 Thread-54 :: Failed doing webui callback: Traceback (most recent call last):
  File "/opt/sickbeard/sickbeard/webserve.py", line 274, in async_call
    result = function(**kwargs)
TypeError: index() got an unexpected keyword argument 'cmd'

If I leave the settings and try to access "Shows", "Coming Episodes" or "History" in the apps nothing is displayed and I got some error messages in the log on the SickRage server: 

2015-09-27 18:00:33 Thread-52 :: Failed doing webui callback: Traceback (most recent call last):
  File "/opt/sickbeard/sickbeard/webserve.py", line 274, in async_call
    result = function(*_kwargs)
  File "/opt/sickbeard/sickbeard/webserve.py", line 681, in index
    t = PageTemplate(rh=self, file="home.mako")
  File "/opt/sickbeard/sickbeard/webserve.py", line 115, in **init**
    super(PageTemplate, self).**init**(_args, **kwargs)
  File "/opt/sickbeard/lib/mako/template.py", line 321, in **init**
    module = self._compile_from_file(path, filename)
  File "/opt/sickbeard/lib/mako/template.py", line 379, in _compile_from_file
    module = compat.load_module(self.module_id, path)
  File "/opt/sickbeard/lib/mako/compat.py", line 66, in load_module
    return imp.load_source(module_id, path, fp)
ImportError: Loaded module _opt_sickbeard_gui_slick_views_home_mako not found in sys.modules

It seems like it is a problem with the SickRage server isn't it?

**Comment by MGaetan89** [comment]
I should make the error message more explicit, because it is not really helpful for the moment.
Can you tell me which version of ShowsRage you are using? You can find this info in Settings > ShowsRage.
Are you using HTTPS? Self-signed certificate? Basic authentication?

The second stacktrace is related to SickRage. They have an issue open for that: https://github.com/SiCKRAGETV/sickrage-issues/issues/2781

**Comment by benwut** [comment]
ShowsRage version 1.0.1
No, I am not using HTTPS neither Self-signed certificate or user/password authentication.

I've tried to put a user/password authentication but the problem is still.

edit: after reading to open ticket on SiCKRAGETV: I am using SickRage on openmediavault. They are speaking that this very problem apprears because of the path of the cache. And I am using it on my openmediavault server so maybe it's where the problem comes from?

**Comment by MGaetan89** [comment]
It looks like you didn't set the path or API key. Is that correct or did you just remove them from the issue?

The second part is related to the cache folder problem, yes. But I don't think that it affects access to SickRage.

**Comment by benwut** [comment]
So yes there is something I don't really get abouth the API thing. First I've tried to connect fulfilling only "Address" and "port number" which are correct (I let the "API" in the path field). Did not work.
Then I've tried to generate an API Key on the SickRage web interface and past it in the App but it didn't work neither. 

Am I doing something wrong? I am not sure about what I need to do with the API Key :@

**Comment by benwut** [comment]
Hoooooooolly shit you can close the ticket. 
It's OK I made mistake copy/pasting the API Key ... I feel so ashamed damn it! Sorry again!

**Comment by MGaetan89** [comment]
The API key is used to allow third party application to access your SickRage installation.
If you managed to generate it in SickRage, that's a good start.

This is how I configured ShowsRage on my phone:
- I set "Address"
- I set "Port number"
- I leave "Path" to "api"
- In "API Key" > "Get API Key":
  - I set the username
  - I set the password
  - I click "Get API Key"

The "username" and "password" are the one set in SickRage > Config > General > Interface > "HTTP username"/"HTTP password".

Edit: no problem. Glad it works. This comment can still be useful as it explains how to automatically get the API key ;)
